client.go

```diff
@@ -483,6 +485,23 @@ func (c *Client) createComponents(config *Config, addrProvider icluster.AddressP
 	c.invocationHandler = invocationHandler
 }
 
+func (c *Client) clusterDisconnected(e event.Event) {
+	if atomic.LoadInt32(&c.state) != ready {
+		return
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), c.clusterConfig.ConnectionTimeout)
```

Comment:
Why do we have this timeout here? Isn't it related to a single connection's timeout?

Reply:
You are right, I've used connection timeout as both the socket connection timeout and (erroneously) as the cluster connection timeout. I'm fixing that in the wrap up PR which includes the connection strategy stuff.